crates/https/src/https_server.rs

```diff
@@ -26,15 +26,15 @@ use crate::HttpsError;
 /// To allow downstream clients to do something interesting with the lifetime of the bytes, this doesn't
 ///   perform a conversion to a Message, only collects all the bytes.
 pub async fn message_from<R>(
-    this_server_name: Arc<String>,
+    this_server_name: Arc<str>,
     request: Request<R>,
 ) -> Result<BytesMut, HttpsError>
 where
     R: Stream<Item = Result<Bytes, h2::Error>> + 'static + Send + Debug + Unpin,
 {
     debug!("Received request: {:#?}", request);
 
-    let this_server_name: &String = this_server_name.borrow();
+    let this_server_name: &str = this_server_name.borrow();
```

Comment:
Can we just get rid of the type annotation here?

Reply:
Oh, this one maybe.